### ISSUE TITLE
crosstalk::filter_select() and colors for plotly pie/donuts

### ISSUE THREAD

**Issue by robertmitchellv** [ropensci/plotly#1503]
First, I want to say thank you for all your hard work. The R API for `plotly` is wonderful and I think it's a fantastic package and tell everyone to use it! Keep up the great work!

Now, the strange behavior that feels buggy. I've been running into strange behavior with pie/donut charts and `plotly`. I'm building `flexdashboards` for small projects at work (folks really want donut charts) and hitting a wall. Some projects have specific color palettes, so I'm trying to incorporate those into the plots, but I'm seeing some strange behavior that doesn't seem expected.

When I use both `crosstalk::filter_select()` with `plotly::plot_ly(markers = list(colors = colors))`, I am not able to get the colors to stick. The example below is to illustrate the problem where the orange and the blue default colors seem to creep back in and I'm not sure why.

```r
library(tidyverse)
library(crosstalk)
library(plotly)

cars <- mtcars %>%
  rownames_to_column("car") %>%
  mutate(car = str_split_fixed(car, " ", 2)[,1]) %>%
  group_by(car) %>%
  summarise(count = n())

shared_cars <- SharedData$new(cars)
colors <- c('rgb(211,94,96)', 'rgb(128,133,133)', 'rgb(144,103,167)', 'rgb(171,104,87)', 'rgb(114,147,203)')

bscols(widths = c(3,NA),
  list(
    filter_select(id = "car_select", label = "Select cars", sharedData = shared_cars, ~car)
  ),
  shared_cars %>% 
    plot_ly(labels = ~car, values = ~count) %>%
    add_pie(
      hole = 0.4,
      marker = list(
        colors = colors
      )
    ) %>%
    layout(title = "Donut charts using Plotly",  showlegend = F,
           xaxis = list(showgrid = FALSE, zeroline = FALSE, showticklabels = FALSE),
           yaxis = list(showgrid = FALSE, zeroline = FALSE, showticklabels = FALSE))
)

```

If I've done anything wrong with creating this issue let me know and I'll make corrections. Thank you!

**Comment by cpsievert** [comment]
Thanks for the kind words @robertmitchellv! 

> I am not able to get the colors to stick. The example below is to illustrate the problem where the orange and the blue default colors seem to creep back in and I'm not sure why.

It'd be helpful to know more specifically what you mean here. This is what I see in the default view with the dev version of plotly -- is this different from what you see?

<img width="364" alt="Screen Shot 2019-03-27 at 2 46 56 PM" src="https://user-images.githubusercontent.com/1365941/55107482-6ec5f500-509f-11e9-889f-cfc5777e6052.png">

It seems as though `marker.color` isn't working because you haven't supplied a length that's equal to the number of categories. If you use `layout.colorway` instead, it will be smarter about repeating colors:

```r
shared_cars %>% 
    plot_ly(labels = ~car, values = ~count) %>%
    add_pie(hole = 0.4) %>%
    layout(title = "Donut charts using Plotly",  showlegend = F,
           xaxis = list(showgrid = FALSE, zeroline = FALSE, showticklabels = FALSE),
           yaxis = list(showgrid = FALSE, zeroline = FALSE, showticklabels = FALSE), colorway = c('rgba(211,94,96,1)', 'rgba(128,133,133,1)', 'rgba(144,103,167,1)', 'rgba(171,104,87,1)', 'rgba(114,147,203,1)'))
```

<img width="382" alt="Screen Shot 2019-03-27 at 2 51 54 PM" src="https://user-images.githubusercontent.com/1365941/55107681-e72cb600-509f-11e9-98ba-7cebdd365b5b.png">

**Comment by robertmitchellv** [comment]
What I meant was that I would get a few of the colors that were correct but then it would begin using orange and blue even before it had exhausted the list of colors that were supplied.

I think I used the wrong example, in my dashboard for work I only have `yes` and `no` as possible responses for my donut charts and I was able to get the supplied colors working when the drop down was not selecting a value, but then when a value was selected the colors would revert to orange and blue.

I just tried `colorway` and it worked. I didn't know this was an option!

I should close this, sorry about that 👍
